piptools/writer.py

```diff
@@ -124,9 +132,12 @@ def write(self, results, reverse_dependencies, primary_packages, markers, hashes
                     f.write(unstyle(line).encode('utf-8'))
                     f.write(os.linesep.encode('utf-8'))
 
-    def _format_requirement(self, ireq, reverse_dependencies, primary_packages, marker=None, hashes=None):
-        line = format_requirement(ireq, marker=marker)
+    def _format_requirement(self, ireq, reverse_dependencies, primary_packages, **kwargs):
```

Comment:
For the sake of argument it makes it harder for discoverability but it makes usage much clearer because you aren't allowing for usages to rely on position.

In terms of discoverability of arguments there should be documentation on the function with these values.